app/vmctl/native/client.go

```diff
@@ -35,39 +41,90 @@ type Response struct {
+	if exploreChunk == stepper.StepHour ||
+		exploreChunk == stepper.StepMinute ||
+		exploreChunk == stepper.StepDay ||
+		exploreChunk == "" {
+		// the minimal step wil be used for metrics explore process
+		exploreChunk = stepper.StepWeek
 	}
 
-	params := req.URL.Query()
-	if f.TimeStart != "" {
-		params.Set("start", f.TimeStart)
+	start, err := utils.GetTime(f.TimeStart)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse time start for explore metrics: %s", err)
 	}
-	if f.TimeEnd != "" {
-		params.Set("end", f.TimeEnd)
+	end, err := utils.GetTime(f.TimeEnd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse time end for explore metrics: %s", err)
 	}
-	params.Set("match[]", f.Match)
-	req.URL.RawQuery = params.Encode()
 
-	resp, err := c.do(req, http.StatusOK)
+	ranges, err := stepper.SplitDateRange(start, end, exploreChunk, false)
 	if err != nil {
-		return nil, fmt.Errorf("series request failed: %s", err)
-	}
 
-	var response Response
-	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
-		return nil, fmt.Errorf("cannot decode series response: %s", err)
-	}
+		return nil, fmt.Errorf("failed to create date ranges for explore metrics: %w", err)
+	}
+
+	var metricNames []string
+	errs, ctx := errgroup.WithContext(ctx)
+	metricNamesC := make(chan []string)
+	for _, times := range ranges {
+		start := times[0].Format(time.RFC3339)
+		end := times[1].Format(time.RFC3339)
+		errs.Go(func() error {
+			url := fmt.Sprintf("%s/%s", c.Addr, nativeMetricNamesAddr)
+			if tenantID != "" {
+				url = fmt.Sprintf("%s/select/%s/prometheus/%s", c.Addr, tenantID, nativeMetricNamesAddr)
+			}
+			req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+			if err != nil {
+				return fmt.Errorf("cannot create request to %q: %s", url, err)
+			}
+
+			params := req.URL.Query()
+			if f.TimeStart != "" {
+				params.Set("start", start)
+			}
+			if f.TimeEnd != "" {
+				params.Set("end", end)
+			}
+			params.Set("match[]", f.Match)
+			req.URL.RawQuery = params.Encode()
+
+			resp, err := c.do(req, http.StatusOK)
+			if err != nil {
+				return fmt.Errorf("series request failed: %s", err)
+			}
+
+			var response Response
+			if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
+				return fmt.Errorf("cannot decode series response: %s", err)
+			}
+
+			if err := resp.Body.Close(); err != nil {
+				return fmt.Errorf("cannot close series response body: %s", err)
+			}
+			select {
+			case metricNamesC <- response.MetricNames:
+				return nil
+			// Close out if another error occurs.
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		})
+	}
+	go func() {
+		if err := errs.Wait(); err != nil {
+			log.Printf("error eplore metrics: %s", err)
+		}
+		close(metricNamesC)
+	}()
 
-	if err := resp.Body.Close(); err != nil {
-		return nil, fmt.Errorf("cannot close series response body: %s", err)
+	for mn := range metricNamesC {
+		metricNames = append(metricNames, mn...)
```

Comment:
It looks like duplicates can be checked here without bloating the `metricNames` slice

Reply:
i used map instead of slice it helps to remove duplicates